### PR TITLE
moved Twig_Template::getAttribute() to a function

### PR DESCRIPTION
**Pull request by fabpot**

### CODE REVIEW

File: lib/Twig/Extension/Core.php [hunk 1]
```diff
@@ -1337,3 +1337,188 @@ function twig_array_batch($items, $size, $fill = null)
...
+        if (null === $object) {
+            $message = sprintf('Impossible to invoke a method ("%s") on a null variable.', $item);
+        } else {
+            $message = sprintf('Impossible to invoke a method ("%s") on a %s variable ("%s").', $item, gettype($object), $object);
+        }
+
+        throw new Twig_Error_Runtime($message, -1, $source);
+    }
+
+    if ($object instanceof Twig_Template) {
```
Comment (stof) [inline lib/Twig/Extension/Core.php]: shouldn't it be check first ? No need to check for ArrayAccess on a Twig_Template

**Comment by stof** [comment]
:+1:

File: lib/Twig/Extension/Core.php (hunk 1 above)
Comment (fabpot) [inline lib/Twig/Extension/Core.php]: I've done the contrary, making the check as late as possible as this should happen rarely.

Comment (stof) [inline lib/Twig/Extension/Core.php]: Can you add a test covering it though ? The C extension currently misses this check but passes tests

**Pull request merged**